src/Proto.Persistence/Persistence.cs

```diff
@@ -28,39 +27,39 @@ public class Persistence
 
             if (t != null)
             {
-                SnapshotIndex = t.Item2;
+                Index = t.Item2;
                 await Context.ReceiveAsync(new RecoverSnapshot(t.Item1));
             }
 
-            await State.GetEventsAsync(Name, EventIndex, e =>
+            await State.GetEventsAsync(Name, Index, e =>
             {
                 Context.ReceiveAsync(new RecoverEvent(e)).Wait();
-                EventIndex++;
+                Index++;
             });
             
             await Context.ReceiveAsync(new RecoveryCompleted());
         }
 
-        public async Task PersistReceiveAsync(object data)
+        public async Task PersistEventAsync(object data)
         {
-            var persistEventIndex = EventIndex;
+            var index = Index;
 
-            await State.PersistEventAsync(Name, persistEventIndex, data);
+            await State.PersistEventAsync(Name, index, data);
 
-            EventIndex++;
+            Index++;
 
-            await Context.ReceiveAsync(new PersistedEvent(persistEventIndex, data));
+            await Context.ReceiveAsync(new PersistedEvent(index, data));
         }
 
         public async Task PersistSnapshotAsync(object data)
         {
-            var persistSnapshotIndex = SnapshotIndex;
+            var index = Index;
 
-            await State.PersistSnapshotAsync(Name, persistSnapshotIndex, data);
+            await State.PersistSnapshotAsync(Name, index, data);
 
-            SnapshotIndex++;
+            Index++;

```

Comment:
I've not tested yet, but I don't think saving a snapshot should increment the event index. For example, I take a snapshot at event index 10, then 5 minutes pass and I take another snapshot, but I received no new events during this time, the event index should be the same, 10.

Another example, if I take 20 snapshots at event index 10, they are all the same capture of the state of the actor at that index. The event index shouldn't now be 30 as no new events have been received.